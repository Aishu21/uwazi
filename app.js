require("node-jsx").install();
var express = require('express');
var app = express();

var react = require('react');
var router = require('react-router');

var renderToString = require('react-dom/server').renderToString;
var routes = require('./js/routes.jsx');
var match = require('react-router').match;
var RoutingContext = require('react-router').RoutingContext;
//var reactApp = React.createFactory(require('./js/react_welcome.js'));
//var layout = require('./js/layout.jsx');

app.get('/', function (req, res) {

  match({ routes: routes, location: req.url }, function(error, redirectLocation, renderProps) {
    res.status(200).send(renderToString(new RoutingContext(renderProps)))
  });

  //res.render(__dirname+'/views/index.ejs', {
    //content: content
  //});

  //router.run(routes, '/', function (Handler) {
    //var content = react.renderToString(react.createElement(Handler));
    //res.render(__dirname+'/views/index.ejs', {
      //content: content
    //});
  //});
})

//app.get('/', function (req, res) {
  //var reactHtml = React.renderToString(ReactApp({}));
  //res.render(__dirname+'/views/welcome.ejs', {reactOutput: reactHtml});
//});

var server = app.listen(3000, function () {
  var host = server.address().address;
  var port = server.address().port;

  console.log('Example app listening at http://%s:%s', host, port);
});
