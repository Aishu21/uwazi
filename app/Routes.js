import React from 'react'
import { Route, IndexRoute } from 'react-router'
import App from './components/App'
import Home from './components/Home'
import Users from './components/Users'
import Login from './components/Login'
import NoMatch from './components/NoMatch'

export default (
	<Route path='/' component={App}>
		<IndexRoute component={Home} />
		<Route path='users' component={Users} />
		<Route path='login' component={Login} />
		<Route path="*" component={NoMatch} />
	</Route>
);
