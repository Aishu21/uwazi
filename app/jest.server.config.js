module.exports = {
  name: 'server',
  displayName: 'Server',
  testMatch: ['**/api/**/specs/*spec.(j|t)s?(x)', '**/shared/**/specs/*spec.(j|t)s?(x)'],
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/setUpJestServer.js'],
  collectCoverageFrom: ['app/**/*.{js,ts}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['none'],
};
