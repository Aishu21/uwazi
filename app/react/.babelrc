{
  "presets": ["es2015", "react"],
  "plugins": [
    ["babel-plugin-module-alias", [
      { "src": "./app/react", "expose": "~" },
      { "src": "./app/shared", "expose": "shared" },
    ]],
    "transform-class-properties",
    "add-module-exports"
  ]
}
