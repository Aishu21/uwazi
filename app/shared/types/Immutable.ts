/** @format */

export type IImmutable<T> = T extends string
  ? string
  : T extends number
  ? number
  : T extends boolean
  ? boolean
  : T extends null
  ? null
  : T extends undefined
  ? undefined
  : T extends Array<infer Elem>
  ? {
      toJS(): Elem[];
      get(i: number): IImmutable<Elem>;
    }
  : {
      toJS(): T;
      get<Field extends keyof T>(_field: Field): IImmutable<T[Field]>;
    };
