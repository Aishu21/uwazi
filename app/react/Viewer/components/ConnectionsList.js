import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { t } from 'app/I18N';
import { Icon } from 'UI';

import { closePanel, deactivateReference } from 'app/Viewer/actions/uiActions';
import Connection from './Connection';

import 'app/Viewer/scss/viewReferencesPanel.scss';

export class ConnectionsList extends Component {
  close() {
    this.props.closePanel();
    this.props.deactivateReference();
  }

  componentWillReceiveProps(newProps) {
    Object.keys(newProps).map((key) => {
      if (newProps[key] !== this.props[key]) {
        console.log(key, newProps[key], this.props[key]);
      }
    });
  }

  render() {
    const references = this.props.references.toJS().sort((a, b) => {
      const aStart = typeof a.range.start !== 'undefined' ? a.range.start : -1;
      const bStart = typeof b.range.start !== 'undefined' ? b.range.start : -1;
      return aStart - bStart;
    });

    if (this.props.loading) {
      return false;
    }

    if (!this.props.references.size && this.props.referencesSection === 'references') {
      return (
        <div className="blank-state">
          <Icon icon="sitemap" />
          <h4>{t('System', 'No References')}</h4>
          <p>{t('System', 'No References description')}</p>
        </div>
      );
    }

    if (!this.props.references.size) {
      return (
        <div className="blank-state">
          <Icon icon="sitemap" />
          <h4>{t('System', 'No Connections')}</h4>
          <p>{t('System', 'No Connections description')}</p>
        </div>
      );
    }

    return (
      <div className="item-group">
        {(() => references.map((reference, index) => <Connection key={index} readOnly={this.props.readOnly} reference={reference} />))()}
      </div>
    );
  }
}
ConnectionsList.defaultProps = {
};


ConnectionsList.propTypes = {
  references: PropTypes.object,
  deactivateReference: PropTypes.func,
  closePanel: PropTypes.func,
  loading: PropTypes.bool,
  readOnly: PropTypes.bool,
  referencesSection: PropTypes.string,
};

ConnectionsList.contextTypes = {
  confirm: PropTypes.func
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ closePanel, deactivateReference }, dispatch);
}

export default connect(null, mapDispatchToProps)(ConnectionsList);
