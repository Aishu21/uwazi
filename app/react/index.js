import React from 'react';
import ReactDOM from 'react-dom';
import {browserHistory} from 'react-router';
import {Router} from 'react-router';
import Routes from './Routes';
import {Provider} from 'react-redux';
import CustomProvider from './App/Provider';
import store from './store';
import {AppContainer} from 'react-hot-loader';

const App = () => {
  return (
    <Provider store={store()}>
      <CustomProvider>
        <Router history={browserHistory}>{Routes}</Router>
      </CustomProvider>
    </Provider>
  );
};

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <App/>
    </AppContainer>,
    document.getElementById('root')
  );
};

render();
if (module.hot) {
  module.hot.accept();
}

import './sockets';

