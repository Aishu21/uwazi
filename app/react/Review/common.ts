/** @format */

import { IImmutable } from 'shared/interfaces/Immutable.interface';
import { EntitySchema } from 'api/entities/entityType';
import { TemplateSchema } from 'api/templates/templateType';
import { ThesaurusSchema } from 'api/thesauris/dictionariesType';

export interface OneUpState {
  loaded: boolean;
  fullEdit: boolean;
  loadConnections: boolean;
  indexInDocs: number;
  totalDocs: number;
  maxTotalDocs: number;
  requestHeaders: Object;
  reviewThesaurusName: string | null;
  reviewThesaurusId: string | null;
  reviewThesaurusValues: string[];
}

export interface StoreState {
  relationships: any;
  entityView: {
    entity: IImmutable<EntitySchema>;
    entityFormState: any;
    entityForm: any;
    uiState: IImmutable<{ tab: string }>;
  };
  oneUpReview: {
    state?: IImmutable<OneUpState>;
  };
  templates: IImmutable<TemplateSchema[]>;
  thesauris: IImmutable<ThesaurusSchema[]>;
  library: {
    documents: IImmutable<{ rows: EntitySchema[] }>;
  };
}
