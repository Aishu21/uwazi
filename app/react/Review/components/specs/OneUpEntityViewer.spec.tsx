/** @format */

import { ConnectionsList } from 'app/ConnectionsList';
import { shallow, ShallowWrapper } from 'enzyme';
import Immutable from 'immutable';
import React from 'react';
import { OneUpState } from '../../common';
import {
  OneUpEntityViewerBase,
  OneUpEntityViewerProps,
  OneUpEntityViewerState,
} from '../OneUpEntityViewer';

describe('EntityViewer', () => {
  let component: ShallowWrapper<
    OneUpEntityViewerProps,
    OneUpEntityViewerState,
    OneUpEntityViewerBase
  >;
  let props: OneUpEntityViewerProps;
  let context: any;

  beforeEach(() => {
    context = { confirm: jasmine.createSpy('confirm') };
    props = {
      entity: { title: 'Title' },
      templates: Immutable.fromJS([
        {
          _id: 'template1',
          properties: [{ name: 'source_property', label: 'label1' }],
          name: 'template1Name',
        },
        {
          _id: 'template2',
          properties: [{ name: 'source_property', label: 'label2' }],
          name: 'template2Name',
        },
      ]),
      relationships: Immutable.fromJS([]),
      connectionsGroups: Immutable.fromJS([
        { key: 'g1', templates: [{ _id: 't1', count: 1 }] },
        {
          key: 'g2',
          templates: [
            { _id: 't2', count: 2 },
            { _id: 't3', count: 3 },
          ],
        },
      ]),
      selectedConnection: false,
      tab: 'info',
      oneUpState: {} as OneUpState,
      deleteConnection: jasmine.createSpy('deleteConnection'),
      showTab: jasmine.createSpy('showTab'),
      connectionsChanged: jasmine.createSpy('connectionsChanged'),
      switchOneUpEntity: jasmine.createSpy('switchOneUpEntity'),
      toggleOneUpFullEdit: jasmine.createSpy('toggleOneUpFullEdit'),
      toggleOneUpLoadConnections: jasmine.createSpy('toggleOneUpLoadConnections'),
      mlThesauri: [],
    };
  });

  const render = () => {
    component = shallow(<OneUpEntityViewerBase {...props} />, { context });
  };

  it('should render', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  it('should render the ConnectionsList passing deleteConnection as prop', () => {
    render();

    component
      .find(ConnectionsList)
      .props()
      .deleteConnection({ sourceType: 'not metadata' });
    expect(context.confirm).toHaveBeenCalled();
  });

  describe('deleteConnection', () => {
    beforeEach(() => {
      render();
    });

    it('should confirm deleting a Reference', () => {
      component.instance().deleteConnection({});
      expect(context.confirm).toHaveBeenCalled();
      expect(props.deleteConnection).not.toHaveBeenCalled();
    });

    it('should delete the reference upon accepting', () => {
      const ref = { _id: 'r1' };
      component.instance().deleteConnection(ref);
      context.confirm.calls.argsFor(0)[0].accept();
      expect(props.deleteConnection).toHaveBeenCalledWith(ref);
    });

    it('should not atempt to delete references whos source is metadata', () => {
      const ref = { _id: 'r1', sourceType: 'metadata' };
      component.instance().deleteConnection(ref);
      expect(context.confirm).not.toHaveBeenCalled();
      expect(props.deleteConnection).not.toHaveBeenCalled();
    });
  });

  describe('closing side panel', () => {
    beforeEach(() => {
      render();
      component.find('.closeSidepanel').simulate('click');
      component.update();
    });
    it('should close the side panel when close button is clicked', () => {
      expect(component.find('.entity-info').prop('open')).toBe(false);
      expect(component.find('.entity-connections').prop('open')).toBe(false);
      expect(component.find('.show-info-sidepanel-context-menu').prop('show')).toBe(true);
    });
    it('should reveal side panel when context menu is clicked', () => {
      expect(component.find('.entity-info').prop('open')).toBe(false);

      (component.find('.show-info-sidepanel-menu').prop('openPanel') as () => {})();
      component.update();

      expect(component.find('.entity-info').prop('open')).toBe(true);
      expect(component.find('.show-info-sidepanel-context-menu').prop('show')).toBe(false);
    });
  });
});
