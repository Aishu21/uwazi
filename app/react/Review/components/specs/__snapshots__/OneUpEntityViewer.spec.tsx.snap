// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityViewer should render 1`] = `
<div
  className="row flex"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Title"
  />
  <div
    className="content-holder"
  >
    <main
      className="content-main"
    >
      <div
        className="content-header content-header-entity"
      >
        <Connect(StateSelectorBase)
          isPristine={[Function]}
        >
          <Component />
        </Connect(StateSelectorBase)>
        <button
          className="btn btn-default btn-toggle-off"
          onClick={[Function]}
          type="button"
        >
          <Connect(Icon)
            icon="toggle-off"
          />
          <span
            className="btn-label"
          >
            Full edit mode
          </span>
        </button>
      </div>
      <div
        className="entity-viewer"
      >
        <Tabs
          selectedTab="info"
        >
          <TabContent
            for="info"
          >
            <div
              className="entity-metadata"
            >
              <ShowIf>
                <Connect(MetadataForm)
                  id="fullEditMetadataForm"
                  model="entityView.entityForm"
                  showSubset={
                    Array [
                      "title",
                    ]
                  }
                  templateId=""
                  version="OneUp"
                />
              </ShowIf>
              <ShowIf
                if={true}
              >
                <div>
                  <div
                    className="content-header-title"
                  >
                    <Connect(Icon)
                      className="item-icon item-icon-center"
                      size="sm"
                    />
                    <h1
                      className="item-name"
                    >
                      Title
                    </h1>
                    <Connect(TemplateLabel)
                      template=""
                    />
                    <Tip
                      icon="eye-slash"
                    >
                      This entity is not public.
                    </Tip>
                  </div>
                  <Connect(ShowMetadata)
                    entity={
                      Object {
                        "title": "Title",
                      }
                    }
                    relationships={Immutable.List []}
                    showSubset={Array []}
                    showTitle={false}
                    showType={false}
                  />
                  <Connect(AttachmentsList)
                    entityView={true}
                    files={Immutable.List []}
                    isDocumentAttachments={false}
                  />
                </div>
              </ShowIf>
            </div>
          </TabContent>
          <TabContent
            for="connections"
          >
            <Connect(withRouter(DocumentsList))
              deleteConnection={[Function]}
              hideFooter={true}
              searchCentered={true}
            />
          </TabContent>
        </Tabs>
      </div>
      <ShowIf
        if={false}
      >
        <div
          className="content-footer"
        >
          <Connect(RelationshipsFormButtons) />
        </div>
      </ShowIf>
      <ShowIf
        if={true}
      >
        <Connect(StateSelectorBase)
          isPristine={[Function]}
        >
          <Component />
        </Connect(StateSelectorBase)>
      </ShowIf>
      <Connect(ContextMenu)
        align="bottom"
        className="show-info-sidepanel-context-menu"
        overrideShow={true}
        show={false}
      >
        <ShowSidepanelMenu
          active={false}
          className="show-info-sidepanel-menu"
          openPanel={[Function]}
          panelIsOpen={true}
        />
      </Connect(ContextMenu)>
    </main>
    <SidePanel
      className="entity-connections entity-info"
      open={true}
    >
      <div
        className="sidepanel-header content-header-tabs"
      >
        <div
          className="blank"
        />
        <Tabs
          handleSelect={[Function]}
          selectedTab="info"
        >
          <ul
            className="nav nav-tabs"
          >
            <li>
              <TabLink
                to="info"
              >
                <Connect(Icon)
                  icon="info-circle"
                />
                <span
                  className="tab-link-tooltip"
                >
                  Info
                </span>
              </TabLink>
            </li>
            <li>
              <TabLink
                to="connections"
              >
                <Connect(Icon)
                  icon="exchange-alt"
                />
                <span
                  className="connectionsNumber"
                >
                  6
                </span>
                <span
                  className="tab-link-tooltip"
                >
                  Connections
                </span>
              </TabLink>
            </li>
          </ul>
        </Tabs>
        <button
          className="closeSidepanel close-modal"
          onClick={[Function]}
          type="button"
        >
          <Connect(Icon)
            icon="times"
          />
        </button>
      </div>
      <div
        className="sidepanel-body"
      >
        <Tabs
          selectedTab="info"
        >
          <TabContent
            for="none"
          >
            <Connect(ConnectionsGroups) />
          </TabContent>
          <TabContent
            for="info"
          >
            <Connect(MetadataForm)
              id="sidePanelMetadataForm"
              model="entityView.entityForm"
              showSubset={Array []}
              templateId=""
              version="OneUp"
            />
          </TabContent>
        </Tabs>
      </div>
      <ShowIf
        if={false}
      >
        <Connect(StateSelectorBase)
          isPristine={[Function]}
        >
          <Component />
        </Connect(StateSelectorBase)>
      </ShowIf>
    </SidePanel>
    <Connect(CreateConnectionPanel)
      className="entity-create-connection-panel"
      onCreate={[Function]}
    />
    <Connect(AddEntities) />
    <Connect(RelationshipMetadata) />
  </div>
  <Connect(Footer) />
</div>
`;
